File: work.flow.one/src/main/java/com/touchmind/work/flow/one/dto/UserResponse.java
package com.touchmind.work.flow.one.dto;

import java.time.Instant;
import java.time.LocalDate;
import java.util.Map;
import java.util.Set;

public record UserResponse(

        String id,

        String username,

        String email,

        String fullName,

        Set<String> roles,

        boolean enabled,

        boolean active,

        Instant createdAt,

        String phoneNumber,

        LocalDate birthday,

        String position,

        String profilePictureUrl,

        Map<String, String> socialContacts

) {
}

File: work-flow-one-ui/src/types/UserResponse.ts
export type UserResponse = {
  id: string;
  username: string;
  email: string;
  fullName: string | null;
  roles: string[];
  enabled: boolean;
  active: boolean;
  createdAt: string;
  phoneNumber: string | null;
  birthday: string | null;
  position: string | null;
  profilePictureUrl: string | null;
  socialContacts: Record<string, string>;
};
